### ISSUE TITLE
'mail' from mailutils hardcodes /usr/sbin/sendmail

### ISSUE THREAD

**Issue by bjornfor** [NixOS/nixpkgs#21008]
... rendering the 'mail' command not working on NixOS.

Last bug report for today :-)

**Comment by mogria** [comment]
Well which sendmail should be used then? `sendmail` is provided by `busybox`, `exim`, `postfix` & `ssmtp`.
Should the installed one be used?

**Comment by bjornfor** [comment]
@mogria: I'd go for $PATH lookup at runtime.

**Comment by peterhoeg** [comment]
Can I ask for this bug to be left alone until tomorrow as I'm giving a talk on NixOS and will be fixing this as part of it?

**Comment by mogria** [comment]
i wasn't able to fix it, so go on :)

On 6 Apr 2017 11:16, "Peter Hoeg" <notifications@github.com> wrote:

Can I ask for this bug to be left alone until tomorrow as I'm giving a talk
on NixOS and will be fixing this as part of it?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/NixOS/nixpkgs/issues/21008#issuecomment-292116033>, or mute
the thread

**Comment by unode** [comment]
Any fix in the works? Just hit this seeing:
```
Cannot open mailer: No such file or directory
cannot send message: No such file or directory
```

**Comment by unode** [comment]
Also am I wrong in interpreting this as a regression of #3619 ?

**Comment by bjornfor** [comment]
Oh, I didn't know we previously had hardcoded the sendmail path. Apparently that was removed in 27a9ef8709bf0fe257a0e9acd13bb6b71d1244be.

**Comment by bjornfor** [comment]
@vrthra: Do you remember why you removed sendmailPath from mailutils? (Commit message doesn't say.)

**Comment by bjornfor** [comment]
I asked upstream about looking up sendmail in $PATH: https://www.mail-archive.com/bug-mailutils@gnu.org/msg01740.html

**Comment by vrthra** [comment]
@bjornfor I think this was a mistake in my commit.

**Comment by unode** [comment]
I can confirm that manually applying the changes introduced in #3619 works around the `Cannot open mailer` problem.

Having `postfix` configured locally I then ran into a problem of `/var/spool/mail/user` being a file instead of a folder (mbox vs maildir format?) leading to failed deliveries.
Removing the location manually and re-sending a message to this user allowed `postfix` to recreate the location in the correct format, solving the issue.

**Comment by unode** [comment]
Updated patch https://gist.github.com/cc29b145ddd6f45ae5701b4d31796c8b

**Comment by bjornfor** [comment]
Fixed by b2b44c9e9fbcfe1215cceec68a062e1fbbd80710.
